src/App.js

```diff
@@ -1,24 +1,44 @@
 import React from 'react';
 import { BrowserRouter as Router, Route } from 'react-router-dom';
+import { USER_TOKEN } from './constants';
+import getToken from './tokenScript';
+import NavButton from './components/navbutton';
+import Shopping from './components/shopping';
+import AddItem from './components/addItem';
+import { useToken } from './lib/useToken';
 import './App.css';
-import NavButton from './components/navbutton.js';
-import Shopping from './components/shopping.js';
-import AddItem from './components/addItem.js';
 
 function App() {
-  return (
-    <Router>
-      <div className="App">
-        <Route exact path="/" component={Shopping} />
-        <Route path="/add" component={AddItem} />
+  const [userToken, setToken] = useToken();
+
+  function handleClick() {
+    const token = getToken();
+    localStorage.setItem(USER_TOKEN, token);
+    setToken(token);
+  }
 
-        <nav>
-          <NavButton path="/" text="Shopping" />
-          <NavButton path="/add" text="Add Item" />
-        </nav>
+  if (userToken) {
```

Comment:
To solve the `NavButtons` showing when there is no token we reorganized our components, moving token logic to `App.js`.